src/MainOpt.cxx

```diff
@@ -120,119 +112,128 @@ int MainOpt::parse_json_file(string config_file) {
+  if (itr != document.MemberEnd() && itr->value.IsString()) {
+    uri::URI u1;
+    u1.port = 9092;
+    u1.parse(itr->value.GetString());
+    status_uri = u1;
   }
-  {
-    auto m1 = d.FindMember("kafka");
-    if (m1 != d.MemberEnd()) {
-      auto &v = m1->value;
-      if (v.IsObject()) {
-        auto m2 = v.FindMember("broker");
-        if (m2 != d.MemberEnd()) {
-          auto &v2 = m2->value;
-          if (v2.IsObject()) {
-            for (auto &x : v2.GetObject()) {
-              auto const &n = x.name.GetString();
-              if (strncmp("___", n, 3) == 0) {
-                // ignore
+  LOG(3, "ERROR could not find status URI in config json")
+}
+
+void MainOpt::find_brokers_config(rapidjson::Document &document) {
+  auto kafka = document.FindMember("kafka");
+  if (kafka != document.MemberEnd()) {
+    auto &brokers = kafka->value;
+    if (brokers.IsObject()) {
+      auto broker = brokers.FindMember("broker");
+      if (broker != document.MemberEnd()) {
+        auto &broker_properties = broker->value;
+        if (broker_properties.IsObject()) {
+          for (auto &broker_property : broker_properties.GetObject()) {
+            auto const &name = broker_property.name.GetString();
+            if (strncmp("___", name, 3) != 0) {
+              if (broker_property.value.IsString()) {
+                auto const &value = broker_property.value.GetString();
+                LOG(6, "kafka broker config {}: {}", name, value);
+                broker_opt.conf_strings[name] = value;
+              } else if (broker_property.value.IsInt()) {
+                auto const &value = broker_property.value.GetUint();
+                LOG(6, "kafka broker config {}: {}", name, value);
+                broker_opt.conf_ints[name] = value;
               } else {
-                if (x.value.IsString()) {
-                  auto const &v = x.value.GetString();
-                  LOG(6, "kafka broker config {}: {}", n, v);
-                  broker_opt.conf_strings[n] = v;
-                } else if (x.value.IsInt()) {
-                  auto const &v = x.value.GetInt();
-                  LOG(6, "kafka broker config {}: {}", n, v);
-                  broker_opt.conf_ints[n] = v;
-                } else {
-                  LOG(3, "ERROR can not understand option: {}", n);
-                }
+                LOG(3, "ERROR can not understand option: {}", name);
               }
             }
           }
         }
       }
     }
   }
-  {
-    auto &v = d.FindMember("status-uri")->value;
-    if (v.IsString()) {
-      uri::URI u1;
-      u1.port = 9092;
-      u1.parse(v.GetString());
-      status_uri = u1;
+}
+
+int MainOpt::find_int(rapidjson::Document &document, const char *key) const {
+  auto itr = document.FindMember(key);
+  if (itr != document.MemberEnd()) {
+    if (itr->value.IsInt()) {
+      return itr->value.GetInt();
     }
   }
+  LOG(3, "ERROR could not find {} in json", key)
```

Comment:
This should not unconditionally print an error. All of the options are technically optional, and things like `conversion-worker-queue-size` is usually not given in the configuration file.